ignite/metrics/gan/inception_score.py

```diff
@@ -0,0 +1,110 @@
+from typing import Callable, Optional, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.gan.utils import InceptionModel
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["InceptionScore"]
+
+
+class InceptionScore(Metric):
+    r"""Calculates Inception Score.
+
+    .. math::
+       \text{IS(G)} = \frac{1}{N}\sum_{i=1}^{N} D_{KL} (p(y|x^{(i)} \parallel \hat{p}(y)))
+
+    where :math:`p(y|x)` is the conditional probability of image being the given object and
+    :math:`p(y)` is the marginal probability that the given image is real.
+
+    More details can be found in `Barratt et al. 2018`__.
+
+    __ https://arxiv.org/pdf/1801.01973.pdf
+
+
+    Args:
+        num_probabilities: number of probabilities predicted by the model or number of classes of the model
+        prediction_model: a callable for predicting the probabilities from the input data. If neither
+            num_probabilities nor prediction_model are defined, default value is ``InceptionModel``.
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            from ignite.metric.gan.IS import InceptionScore
+            import torch
+
+            probabilities = torch.rand(10, 1000)
+
+            m = InceptionScore(num_probabilities=1000)
+            m.update(probabilities)
+            print(m.compute())
```

Comment:
Let's provide GAN-related example instead, where we measure metric value on a random image.